DOCSP-33790 Deployment Considerations Content (#136)
mongodb/docs-relational-migrator: source/deployment-considerations/colocation.txt

.. _rm-colocation:

==========
Colocation
==========

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

> 

Reviewer: I see that the page is titled colcation, but we don't define what that is anywhere on the page, and instead jump right into kind of more general recommendations for environments. Maybe we should add a brief introduction for what colocation is and why it's important?

Reviewer: Nice addition. The context is clearer now.

Reviewer 2: Do we really need a hierarchy here? "Deployment considerations" is a better page title than "Colocation".

> How to run Relational Migrator

Reviewer: I think "Run" should be capitalized in these titles.

------------------------------

The following table outlines some high level recommendations for how to 
run Relational Migrator. Also note that the 
:ref:`Kafka deployment model <kafka-intro>` allows you greater 
resilience and scalability and can be used with any of the 
deployment options.

.. list-table::
   :header-rows: 1
   :widths: 40 60

   * - Use case 
     - Recommended environment

   * - Smaller scale jobs that run for less than 4 hours, general
       debugging and testing.
     - Labtop / Workstation / Docker

   * - Larger scale production jobs that run for more than 4 
       hours.
     - Remote server / EC2 / Azure VM

Where to run Relational Migrator
--------------------------------

For best performance, locate the machine or server running Relational 
Migrator as geographically close to the source database as possible: 

> - If your using Relational Migrator for an on-premises migration, this 

Reviewer: Wrong "your". This "your" is possessive. You want the contraction "you're".

> - If your using Relational Migrator for an on-premises migration, this 
>   means running Relational Migrator in the same data center as the source
>   database.

Reviewer: This wording is a little clunky.

```suggestion
- If your using Relational Migrator for an on-premises migration, run Relational Migrator in the same data center as the source
  database.
```

> - If your using a cloud hosted database this means running Relational 

Reviewer: See above comments for this bullet.

  Migrator on an EC2 instance or VM in the same 
  :abbr:`VPC (Virtual Private Cloud)` as the source database.

.. tip::

   Check the sleep timeout setting for your operating system. If your 

>    machine goes to sleep in the middle of a migration the sync job fails. 

Reviewer: ```suggestion
   machine goes to sleep during migration, the sync job fails. 
```

Network Considerations
----------------------

When using Relational Migrator, the system firewall on the machine or 
server must allow inbound and outbound TCP traffic to the source
database. The source database connection must also allow TCP traffic 
from the IPV4 address of the machine or server running Relational
Migrator.

VPC and Cloud Networking
~~~~~~~~~~~~~~~~~~~~~~~~

When running Relational Migrator on an AWS EC2 instance, check the 
security group and firewall configurations of the EC2 instance or VM. 
Also check that the router table allows traffic to the database. If your 

> database is hosted on a public IP address, this means that the EC2 

Reviewer: ```suggestion
database is hosted on a public IP address, the EC2 
```

instance needs a router table entry to an internet gateway within the 
:abbr:`VPC (Virtual Private Cloud)`. For details on specific cloud 

> provider network configurations see:

Reviewer: Generally we want to introduce lists with a full sentence, so maybe we can try:

```suggestion
provider network configurations see these pages:
```

- `AWS Networking <https://docs.aws.amazon.com/vpc/>`__
- `Azure Networking <https://azure.microsoft.com/en-us/products/category/networking>`__
- `GCP Networking <https://cloud.google.com/vpc/docs>`__

AWS Private Link
~~~~~~~~~~~~~~~~

> `AWS Private Link <https://docs.aws.amazon.com/vpc/latest/privatelink/what-is-privatelink.html>`__ 

Reviewer: Does this blurb need to be so marketing-y?

Reviewer: Thanks, I think this is much better.

> is a robust, scalable solution allowing you to 

Reviewer: ```suggestion
is a robust, scalable solution that allows you to 
```

establish private connections between your VPC and various services, 
making them feel as though they're located within your own VPC. This 

> eliminates the necessity for an internet gateway or public IP addresses.

Reviewer: This last sentence is pretty clunky. What about:

"With AWS Private Link, you don't need an internet gateway or public IP addresses."

You can use Relational Migrator with AWS private link. You need to 
configure a

> `AWS private link endpoint <https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpcendpoint.html>`__.

Reviewer: Consider:

To use Relational Migrator with AWS Private Link, you need to configure...

.. _rm-colocation:

==========
Colocation
==========

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

> In the context of Relational Migrator, colocation refers to where the 

Reviewer 2: As above, I don't think this term is helpful. "Colocation" could imply running multiple things on the same machine, which we don't want to encourage.

servers, networking equipment, and databases are located. This page 
outlines location specific details to consider when using Relational 
Migrator.

How to Run Relational Migrator
------------------------------

> The following table outlines some high level recommendations for how to 

Reviewer: ```suggestion
The following table outlines high-level recommendations for how to 
```

run Relational Migrator. Also note that the 
:ref:`Kafka deployment model <kafka-intro>` provides greater 
resilience and scalability and can be used with any of the 
deployment options.

.. list-table::
   :header-rows: 1
   :widths: 40 60

   * - Use case 
     - Recommended environment

   * - Smaller scale or debugging and testing jobs that run for 
       less than four hours. 
     - Laptop / Desktop / Docker

   * - Larger scale production jobs that run for more than four 
       hours.
     - Remote server / EC2 / Azure VM

> Where to run Relational Migrator

Reviewer: ```suggestion
Where to Run Relational Migrator
```

Author: Woops sorry I missed this from the last round.

--------------------------------

For best performance, locate the machine or server running Relational 

> Migrator as geographically close to the source database as possible: 

Reviewer 2: @alwaysbemark If someone is forced to choose between being close to the source DB and close to the destination DB, is one better than the other?

- If you're using Relational Migrator for an on-premises migration, 
  run Relational Migrator in the same data center as the source 
  database.

> - If you're using a cloud hosted database this means running Relational 
>   Migrator on an EC2 instance or VM in the same 

Reviewer: ```suggestion
- If you're using a cloud hosted database, run Relational 
  Migrator on an EC2 instance or VM in the same 
```

  :abbr:`VPC (Virtual Private Cloud)` as the source database.

.. tip::

   Check the sleep timeout setting for your operating system. If your 
   machine goes to sleep during a migration, the sync job fails. 

Network Considerations
----------------------

When using Relational Migrator, the system firewall on the machine or 

> server must allow inbound and outbound TCP traffic to the source

Reviewer 2: I don't think we need inbound? @alwaysbemark please confirm.

database. The source database connection must also allow TCP traffic 

> from the IPV4 address of the machine or server running Relational

Reviewer 2: Not sure we need to state IPv4. I'd imagine it's possible to get this working on IPv6 if you're keen. Just "IP address" to make it generic?

Migrator.

VPC and Cloud Networking
~~~~~~~~~~~~~~~~~~~~~~~~

> When running Relational Migrator on an AWS EC2 instance, check the 

Reviewer 2: This advice applies to all cloud providers. Can you rewrite it so it doesn't use AWS-specific terminology?

security group and firewall configurations of the EC2 instance or VM. 
Also check that the router table allows traffic to the database. If your 
database is hosted on a public IP address, the EC2 
instance needs a router table entry to an internet gateway within the 
:abbr:`VPC (Virtual Private Cloud)`. For details on specific cloud 
provider network configurations, see these pages:

- `AWS Networking <https://docs.aws.amazon.com/vpc/>`__
- `Azure Networking <https://azure.microsoft.com/en-us/products/category/networking>`__
- `GCP Networking <https://cloud.google.com/vpc/docs>`__

> AWS Private Link

Reviewer 2: I don't think this section adds anything. There isn't anything here specific to RM. There are also equivalent technologies for other cloud vendors.

~~~~~~~~~~~~~~~~

`AWS Private Link <https://docs.aws.amazon.com/vpc/latest/privatelink/what-is-privatelink.html>`__ 

> is a solution that allows you to establish private connections between 

Reviewer: ```suggestion
is a solution that lets you establish private connections between 
```

your VPC and various services. AWS private link makes network traffic 
behave as though the connected systems are located within your own VPC.

You can configure a `AWS private link endpoint <https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpcendpoint.html>`__ 
for usage with Relational Migrator. This setup 
eliminates the need for an internet gateway or public IP addresses.

> System Hardware

Reviewer 2: Somewhere (not necessarily this section), can we call out that migration is I/O intensive on the destination MongoDB cluster, and that scaling up the cluster can significantly reduce migration time?
